src/IO/ADIOS/ADIOS1IOHandler.cpp

```diff
@@ -53,15 +240,1228 @@ ADIOS1IOHandler::flush()
+
+    std::string chunkSize;
+    std::string chunkOffset;
+    int status;
+    for( size_t i = 0; i < ndims; ++i )
+    {
+        chunkSize = "/tmp" + name + "_chunkSize" + std::to_string(i);
+        status = adios_write(fd, chunkSize.c_str(), &parameters.extent[i]);
+        ASSERT(status == err_no_error, "Internal error: Failed to write ADIOS variable during Dataset writing");
+        chunkOffset = "/tmp" + name + "_chunkOffset" + std::to_string(i);
+        status = adios_write(fd, chunkOffset.c_str(), &parameters.offset[i]);
+        ASSERT(status == err_no_error, "Internal error: Failed to write ADIOS variable during Dataset writing");
+    }
+
+    status = adios_write(fd,
+                         name.c_str(),
+                         parameters.data.get());
+    ASSERT(status == err_no_error, "Internal error: Failed to write ADIOS variable during Dataset writing");
+}
+
+void
+ADIOS1IOHandlerImpl::writeAttribute(Writable* writable,
+                                    Parameter< Operation::WRITE_ATT > const& parameters)
+{
+    if( m_handler->accessType == AccessType::READ_ONLY )
+        throw std::runtime_error("Writing an attribute in a file opened as read only is not possible.");
+
+    Attribute att = Attribute(parameters.resource);
+
+    int nelems;
+    switch( parameters.dtype )
+    {
+        using DT = Datatype;
+        case DT::VEC_CHAR:
+            nelems = att.get< std::vector< char > >().size();
+            break;
+        case DT::VEC_INT16:
+            nelems = att.get< std::vector< int16_t > >().size();
+            break;
+        case DT::VEC_INT32:
+            nelems = att.get< std::vector< int32_t > >().size();
+            break;
+        case DT::VEC_INT64:
+            nelems = att.get< std::vector< int64_t > >().size();
+            break;
+        case DT::VEC_UCHAR:
+            nelems = att.get< std::vector< unsigned char > >().size();
+            break;
+        case DT::VEC_UINT16:
+            nelems = att.get< std::vector< uint16_t > >().size();
+            break;
+        case DT::VEC_UINT32:
+            nelems = att.get< std::vector< uint32_t > >().size();
+            break;
+        case DT::VEC_UINT64:
+            nelems = att.get< std::vector< uint64_t > >().size();
+            break;
+        case DT::VEC_FLOAT:
+            nelems = att.get< std::vector< float > >().size();
+            break;
+        case DT::VEC_DOUBLE:
+            nelems = att.get< std::vector< double > >().size();
+            break;
+        case DT::VEC_LONG_DOUBLE:
+            nelems = att.get< std::vector< long double > >().size();
+            break;
+        case DT::VEC_STRING:
+            nelems = att.get< std::vector< std::string > >().size();
+            break;
+        case DT::ARR_DBL_7:
+            nelems = 7;
+            break;
+        case DT::UNDEFINED:
+        case DT::DATATYPE:
+            throw std::runtime_error("Unknown Attribute datatype");
```

Comment:
I am wondering if one has still to `break` after a throw...?
probably, no?

Reply:
Execution at that point aborts until you reach the next mathcing ```catch```. And even inside that catch, you can not possilbly recover to the point of failure (unless you use ```goto```-fuckery).

So no, there is no ```break``` required after a ```throw``` just like after a ```return```.

Reply:
ok, I was more wondering if the `default` can be reached without the `break`.